api/src/test/java/org/pmiops/workbench/workspaces/WorkspacesControllerTest.java

```diff
@@ -1242,6 +1254,121 @@ public void testCloneWorkspaceWithConceptSetNewCdrVersionNewConceptSetCount() th
+    originalConceptSet.setDescription("d1");
+    originalConceptSet.setDomainEnum(Domain.CONDITION);
+    originalConceptSet.setConceptIds(Collections.singleton(CLIENT_CONCEPT_1.getConceptId()));
+    originalConceptSet.setWorkspaceId(dbWorkspace.getWorkspaceId());
+    originalConceptSet = conceptSetDao.save(originalConceptSet);
+
+    org.pmiops.workbench.db.model.Cohort originalCohort =
+        new org.pmiops.workbench.db.model.Cohort();
+    originalCohort.setName("cohort name");
+    originalCohort.setDescription("cohort description");
+    originalCohort.setWorkspaceId(dbWorkspace.getWorkspaceId());
+    originalCohort = cohortDao.save(originalCohort);
+
+    org.pmiops.workbench.db.model.CohortReview originalCohortReview =
+        new org.pmiops.workbench.db.model.CohortReview();
+    originalCohortReview.setCohortName("cohort review");
+    originalCohortReview.setCohortDefinition("cohort definition");
+    originalCohortReview.setCohortId(originalCohort.getCohortId());
+    originalCohortReview = cohortReviewDao.save(originalCohortReview);
+
+    originalCohort.setCohortReviews(Collections.singleton(originalCohortReview));
+    originalCohort = cohortDao.save(originalCohort);
+
+    DataSet originalDataSet = new DataSet();
+    originalDataSet.setName("data set name");
+    originalDataSet.setVersion(1);
+    originalDataSet.setConceptSetIds(
+        Collections.singletonList(originalConceptSet.getConceptSetId()));
+    originalDataSet.setCohortIds(Collections.singletonList(originalCohort.getCohortId()));
+    originalDataSet.setWorkspaceId(dbWorkspace.getWorkspaceId());
+    dataSetDao.save(originalDataSet);
+
+    CloneWorkspaceRequest req = new CloneWorkspaceRequest();
+    Workspace modWorkspace = new Workspace();
+    modWorkspace.setName("cloned");
+    modWorkspace.setNamespace("cloned-ns");
+    modWorkspace.setCdrVersionId(String.valueOf(cdrVersion2.getCdrVersionId()));
+
+    ResearchPurpose modPurpose = new ResearchPurpose();
+    modPurpose.setAncestry(true);
+    modWorkspace.setResearchPurpose(modPurpose);
+    req.setWorkspace(modWorkspace);
+
+    stubGetWorkspace(
+        modWorkspace.getNamespace(),
+        modWorkspace.getName(),
+        LOGGED_IN_USER_EMAIL,
+        WorkspaceAccessLevel.OWNER);
+
+    mockBillingProjectBuffer("cloned-ns");
+    Workspace cloned =
+        workspacesController
+            .cloneWorkspace(workspace.getNamespace(), workspace.getId(), req)
+            .getBody()
+            .getWorkspace();
+
+    org.pmiops.workbench.db.model.Workspace clonedDbWorkspace =
+        workspaceDao.findByWorkspaceNamespaceAndFirecloudNameAndActiveStatus(
+            cloned.getNamespace(),
+            cloned.getId(),
+            StorageEnums.workspaceActiveStatusToStorage(WorkspaceActiveStatus.ACTIVE));
+
+    List<DataSet> dataSets = dataSetService.getDataSets(clonedDbWorkspace);
+    assertThat(dataSets).hasSize(1);
+    assertThat(dataSets.get(0).getName()).isEqualTo("data set name");
```

Comment:
Slight preference for factoring all the magic strings out to the top of the function

Reply:
Will do